library/src/main/java/com/danikula/videocache/HttpUrlSource.java

```diff
@@ -91,6 +95,10 @@ public int read(byte[] buffer) throws ProxyCacheException {
             throw new InterruptedProxyCacheException("Reading source " + url + " is interrupted", e);
         } catch (IOException e) {
             throw new ProxyCacheException("Error reading data from " + url, e);
+        } catch (NullPointerException e) {
```

Comment:
Catching NPE is bad practice. In this case it is just workaround for library bug. So please add some short explanation comments for catch block with link to the [issue](https://github.com/danikula/AndroidVideoCache/issues/29)